File: apps/env_viewer/lib/api/dto/client_update_dto.dart
/// Mirror of the backend `ClientUpdateDto`: the newest published env_viewer
/// build for this platform, with the SHA-256 the client must verify before
/// handing the download to the updater helper.
class ClientUpdateDto {
  final String version;
  final String platform;
  final String fileName;
  final int sizeBytes;
  final String sha256;
  final String? notes;

  const ClientUpdateDto({
    required this.version,
    required this.platform,
    required this.fileName,
    required this.sizeBytes,
    required this.sha256,
    this.notes,
  });

  factory ClientUpdateDto.fromJson(Map<String, dynamic> json) {
    return ClientUpdateDto(
      version: json['version'] as String,
      platform: json['platform'] as String,
      fileName: json['fileName'] as String,
      sizeBytes: (json['sizeBytes'] as num).toInt(),
      sha256: json['sha256'] as String,
      notes: json['notes'] as String?,
    );
  }
}

File: backend/src/main/java/com/flyaif/envdashboard/clientupdate/web/dto/ClientUpdateDto.java
package com.flyaif.envdashboard.clientupdate.web.dto;

import com.flyaif.envdashboard.clientupdate.ClientRelease;

/**
 * The published description of the newest env_viewer build for one platform. {@code sha256} lets
 * the client verify the download before handing it to the updater helper; {@code notes} is the
 * optional operator-written release-notes markdown shown in the update popup.
 */
public record ClientUpdateDto(
        String version,
        String platform,
        String fileName,
        long sizeBytes,
        String sha256,
        String notes
) {

    public static ClientUpdateDto from(ClientRelease release) {
        return new ClientUpdateDto(
                release.version(),
                release.platform(),
                release.zip().getFileName().toString(),
                release.sizeBytes(),
                release.sha256(),
                release.notes());
    }
}
